tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTests.java

```diff
@@ -208,6 +215,52 @@ public void join() throws WebApplicationException {
                 lraClient.isLRAFinished(lra));
     }
 
+    /**
+     * TCK test to verify that methods annotated with {@link AfterLRA}
+     * are notified correctly when an LRA terminates
+     */
+    @Test
+    public void testAfterLRAParticipant() throws WebApplicationException, InterruptedException {
+        URI lra = lraClient.startLRA(null, lraClientId(), lraTimeout(), ChronoUnit.MILLIS);
+        WebTarget resourcePath = tckSuiteTarget.path(AFTER_LRA_PARTICIPANT_PATH).path(AFTER_LRA_PARTICIPANT_WORK);
+        Response response = resourcePath
+                .request().header(LRA_HTTP_CONTEXT_HEADER, lra).put(Entity.text(""));
+        checkStatusAndCloseResponse(Response.Status.OK, response, resourcePath);
+        lraClient.closeLRA(lra);
+
+        // verify that the LRA is now in one of the terminal states
+        assertTrue("testAfterLRAParticipant: LRA did not finish",
+                lraClient.isLRAFinished(lra, lraMetricService, AfterLRAParticipant.class.getName()));
+
+        // verify that the resource was notified of the final state of the LRA
+        assertEquals("testAfterLRAParticipant: end synchronization was not invoked on resource " + resourcePath.getUri(),
+                1,
+                lraMetricService.getMetric(LRAMetricType.Closed, lra, AfterLRAParticipant.class.getName()));
```

Comment:
this test is missing assert that `@AfterLRA` was invoked. Maybe this assert should be checking LRAMetricType.AfterLRA?

Edit: It seems that `@AfterLRA` methods in After LRA listener and participant increment received status metric in their `@AfterLRA` method. But that means that LRAMetricType.Closed should be equal to 2 - once for /complete call and once for /afterLRA call?

Reply:
The only code execution path that increments the Closed metric is the one in the AfterLRA notification method ie this test validates that the resource was told that the LRA had been closed via the `AfterLRA` notification.

Reply:
my bad, mixing Closed and Completed, please resolve